src/main/java/seedu/duke/storage/StorageManager.java

```diff
@@ -0,0 +1,99 @@
+package seedu.duke.storage;
+
+import seedu.duke.model.InternshipApplication;
+import seedu.duke.storage.exceptions.InvalidDelimitedStringException;
+import seedu.duke.storage.exceptions.StorageException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class StorageManager implements Storage {
+    private static final String DEFAULT_FILE_PATH = "data.txt";
+    private final String filePath;
+    private File file;
+
+    public StorageManager(String filePath) {
+        this.filePath = filePath;
+        this.file = new File(filePath);
+    }
+
+    public StorageManager() {
+        this.filePath = DEFAULT_FILE_PATH;
+        file = new File(filePath);
+    }
+
+    @Override
+    public String getFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public ArrayList<InternshipApplication> readApplicationsFromFile()
+            throws StorageException, InvalidDelimitedStringException, FileNotFoundException {
+        requireNonNullFile();
+        Scanner fileScanner = new Scanner(file);
+        ArrayList<InternshipApplication> applicationsList = new ArrayList<>();
+
+        while (fileScanner.hasNextLine()) {
+            String line = fileScanner.nextLine();
+            InternshipApplication application = ApplicationSerializer.delimitedStringToApplication(line);
+            applicationsList.add(application);
+        }
+
+        fileScanner.close();
+        return applicationsList;
+    }
+
+    @Override
+    public void storeApplicationsToFile(InternshipApplication[] applications) throws StorageException {
+        requireNonNullFile();
+        FileWriter fileWriter = null;
+        try {
+            fileWriter = new FileWriter(file);
+        } catch (IOException e) {
+            throw new StorageException("Could not open file " + filePath);
+        }
+
+        // Serialize to storage friendly format, then store
+        for (InternshipApplication application : applications) {
+            String applicationStorageString = ApplicationSerializer.applicationToDelimitedString(application);
+            writeToFile(applicationStorageString, fileWriter);
+        }
+
+        try {
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new StorageException("Could not close file " + file);
+        }
+    }
+
+    private void requireNonNullFile() throws StorageException {
```

Comment:
Not exactly an apt verb for this function name. Something like `checkAndCreateNonNullFile` or `ensureFileExists` would be better.